### PR TITLE
[WASI] platform specific assemblies and fixes for testing

### PR DESCRIPTION
**Pull request by pavelsavara**
- fix `XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS` for windows
- added `OperatingSystem.IsWasi()` to many places
- added `UnsupportedOSPlatformGuard` to few places
- added `<UnsupportedOSPlatforms>wasi</UnsupportedOSPlatforms>` to few places temporarily until https://github.com/dotnet/runtime/pull/107185
- added `throw new PlatformNotSupportedException()` to blocking `.Wait()` places
- fixed xharness runner

### CODE REVIEW

File: eng/testing/xunit/xunit.console.targets [hunk 1]
```diff
@@ -13,8 +13,8 @@
 
   <ItemGroup>
     <!-- Configures xunit to not print out passing tests with output when diagnostic messages are enabled. -->
-    <SetScriptCommands Condition="'$(TargetOS)' == 'windows'" Include="set XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS=1" />
-    <SetScriptCommands Condition="'$(TargetOS)' != 'windows'" Include="export XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS=1" />
+    <SetScriptCommands Condition="'$(OS)' != 'Windows_NT'" Include="set XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS=1" />
+    <SetScriptCommands Condition="'$(OS)' == 'Windows_NT'" Include="export XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS=1" />
```
Comment (ilonatommy) [inline eng/testing/xunit/xunit.console.targets:17]: "fix XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS for windows" means that `TargetOS` was empty so we never triggered  this line? I am wondering why we changed from `TargetOS` to `OS`.

**Review by ilonatommy — APPROVED** [review]

File: eng/testing/xunit/xunit.console.targets (hunk 1 above)
Comment (pavelsavara) [reply eng/testing/xunit/xunit.console.targets:17]: The generated script was wrong and complained to log on linux helix.

This is the change that introduced it https://github.com/dotnet/runtime/pull/105392

File: src/libraries/System.Net.Primitives/src/System.Net.Primitives.csproj [hunk 2]
```diff
@@ -155,7 +155,7 @@
              Link="Common\Interop\Unix\System.Native\Interop.InterfaceNameToIndex.cs" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'browser' or '$(TargetPlatformIdentifier)' == 'wasi'">
+  <ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'browser'">
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/libraries/System.Net.Primitives/src/System.Net.Primitives.csproj:158]: reverted, thanks

[the remark this answers is not in the document]
Comment (ilonatommy) [reply src/libraries/System.Net.Primitives/src/System.Net.Primitives.csproj]: but this one change a few lines above is ok?
`<ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'unix' or '$(TargetPlatformIdentifier)' == 'browser' or '$(TargetPlatformIdentifier)' == 'wasi' ">`

[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/libraries/System.Net.Primitives/src/System.Net.Primitives.csproj]: reverted too

**Pull request merged**